Resources/public/js/views/serverside/ez-roleserversideview.js

```diff
@@ -60,26 +63,86 @@ YUI.add('ez-roleserversideview', function (Y) {
         },
 
         /**
-         * Changes the state of the provided assign button to be *loading* and
+         * tap event handler for policy limitation on location ("Node"). It launches the
+         * universal discovery widget so that the user can pick a location.
+         *
+         * @method _pickLocationLimitation
+         * @protected
+         * @param {EventFacade} e
+         */
+        _pickLocationLimitation: function (e) {
+            var button = e.target,
+                unsetLoading = Y.bind(this._uiUnsetUDWButtonLoading, this, button);
+
+            e.preventDefault();
+            this._uiSetUDWButtonLoading(button);
+            this.fire('contentDiscover', {
+                config: {
+                    title: button.getAttribute('data-universaldiscovery-title'),
+                    cancelDiscoverHandler: unsetLoading,
+                    multiple: true,
+                    contentDiscoveredHandler: Y.bind(this._setLocationLimitation, this, button),
+                },
+            });
+        },
+
+        /**
+         * Puts picked location id into location limitation input. Input is selected by the selector
+         * provided in the `data-location-input-selector` attribute of the button, for example
+         * <button data-location-input-selector="#id_of_input"></button>
+         *
+         * @method _setLocationLimitation
+         * @protected
+         * @param {Y.Node} button
+         * @param {EventFacade} e
+         */
+        _setLocationLimitation: function (button, e) {
```

Comment:
this whole method is very complicated to understand (I had to test it to get what it actually does) because of bad naming and lack of semantic. You should create methods so that the code is self-explanatory.
For instance, this method code should something like:

```js
this._emptyLocationList();
this._emptyLocationIdHiddenInput();

Y.Array.each(e.selection, function (struct) {
    this._addLocationToDisplayList(struct.contentInfo);
    this._addLocationIdToHiddenInput(struct.location.get('locationId'));
}, this);

this._uiUnsetUDWButtonLoading(button);
```